core/sql/optimizer/ScanOptimizer.cpp

```diff
@@ -8703,11 +8713,14 @@ void MDAMCostWA::computeDisjunct()
   if(CURRSTMT_OPTDEFAULTS->indexEliminationLevel() != OptDefaults::MINIMUM
      && (!mdamForced_)
 	 && (CmpCommon::getDefault(RANGESPEC_TRANSFORMATION) == DF_ON )
+	 && (CmpCommon::getDefault(MDAM_APPLY_RESTRICTION_CHECK) == DF_ON )
```

Comment:
The effect of this change is that we will pick MDAM more often? I take it the method "checkMDAMadditionalRestriction" is where the problem was?